From 35fc5db6e6706a101353e08b623e8345a7653e21 Mon Sep 17 00:00:00 2001
From: Prabhakar Kushwaha <prabhakar@freescale.com>
Date: Thu, 14 Jun 2012 11:57:17 +0000
Subject: [PATCH 31/57] t4240: Add IFC support in device tree

commit 6758f1a63f0ae9b22cf4182d6e62bdb8f74e106f from
https://git.freescale.com/git-private/cgit.cgi/ppc/t4240/linux.git

Add support of IFC controller in T4240.
Also Create NOR, NAND flash partitions.

Signed-off-by: Prabhakar Kushwaha <prabhakar@freescale.com>
---
 arch/powerpc/boot/dts/fsl/t4240si-post.dtsi |    7 +++
 arch/powerpc/boot/dts/t4240qds.dts          |   85 +++++++++++++++++++++++++++
 2 files changed, 92 insertions(+)

diff --git a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
index 994c136..5c75206 100644
--- a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
@@ -32,6 +32,13 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+&ifc {
+	#address-cells = <2>;
+	#size-cells = <1>;
+	compatible = "fsl,ifc", "simple-bus";
+	interrupts = <16 2 0 0 25 2 0 0>;
+};
+
 &bportals {
 /include/ "qoriq-bman2-portals.dtsi"
 };
diff --git a/arch/powerpc/boot/dts/t4240qds.dts b/arch/powerpc/boot/dts/t4240qds.dts
index 9489ebc..8449d82 100644
--- a/arch/powerpc/boot/dts/t4240qds.dts
+++ b/arch/powerpc/boot/dts/t4240qds.dts
@@ -41,6 +41,91 @@
 	#size-cells = <2>;
 	interrupt-parent = <&mpic>;
 
+	ifc: localbus@ffe124000 {
+		reg = <0xf 0xfe124000 0 0x1000>;
+		ranges = <0 0 0xf 0xe8000000 0x08000000
+			  2 0 0xf 0xffa00000 0x00010000>;
+
+		nor@0,0 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			compatible = "cfi-flash";
+			reg = <0x0 0x0 0x8000000>;
+			/* TODO: Update bank-width depending upon
+			 * on-board Flash
+			 */
+			bank-width = <1>;
+			device-width = <1>;
+
+			partition@0 {
+				/* This location must not be altered  */
+				/* 1MB for Vitesse 7385 Switch firmware */
+				reg = <0x0 0x00100000>;
+				label = "NOR  Vitesse-7385 Firmware";
+			};
+
+			partition@100000 {
+				/* 1MB for DTB Image */
+				reg = <0x00100000 0x00100000>;
+				label = "NOR  DTB Image";
+			};
+
+			partition@200000 {
+				/* 10 MB for Linux Kernel Image */
+				reg = <0x00200000 0x00A00000>;
+				label = "NOR  Linux Kernel Image";
+			};
+
+			partition@C00000 {
+				/* 115MB for JFFS2 based Root file System */
+				reg = <0x00C00000 0x07300000>;
+				label = "NOR  JFFS2 Root File System";
+			};
+
+			partition@7F00000 {
+				/* This location must not be altered  */
+				/* 512KB for u-boot Bootloader Image */
+				/* 512KB for u-boot Environment Variables */
+				reg = <0x07F00000 0x00100000>;
+				label = "NOR U-Boot Image";
+			};
+		};
+
+		nand@2,0 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			compatible = "fsl,ifc-nand";
+			reg = <0x2 0x0 0x10000>;
+
+			partition@0 {
+				/* This location must not be altered  */
+				/* 1MB for u-boot Bootloader Image */
+				reg = <0x0 0x00100000>;
+				label = "NAND U-Boot Image";
+				read-only;
+			};
+
+			partition@100000 {
+				/* 1MB for DTB Image */
+				reg = <0x00100000 0x00100000>;
+				label = "NAND DTB Image";
+			};
+
+			partition@200000 {
+				/* 10MB for Linux Kernel Image */
+				reg = <0x00200000 0x00A00000>;
+				label = "NAND Linux Kernel Image";
+			};
+
+			partition@C00000 {
+				/* 500MB for Root file System Image */
+				reg = <0x00e00000 0x1F400000>;
+				label = " NAND RFS Image";
+			};
+		};
+
+	};
+
 	memory {
 		device_type = "memory";
 	};
-- 
1.7.9.7

